### ISSUE TITLE
Setting jobname does not play nice with \bibliography command

### ISSUE THREAD

**Issue by sergiocallegari** [latex3/latex2e#189]
When working with bibliographies, it is common to rely on external tools to generate a bbl file that is then automatically loaded on invocation of the `\bibliography` command. When passing latex documents around, this makes it possible to give others the `.tex` source together with a pre-generate `.bbl` file, so saving the need to share multiple `.bib` files and a `.bst` style.

Unfortunately, this breaks when using the `-jobname=` option. In fact, the latter makes latex seek the `.bbl` file as `<jobname>.bbl` where `<jobname>` is the name passed in the `-jobname` option, rather than as `<mainfile>.bbl`, where mainfile is the basename of the latex source file `<mainfile>.tex`.

As a typical example, consider having some document `test.tex`, together with a pre-generated `text.bbl`. When one compiles it as `latex test.tex` everything works. However, doing `latex -jobname=output test.tex` does not work because latex will seek `output.bbl` rather than `test.bbl`.  On one hand this seems reasonable, because `latex -jobname=output test.tex` creates an `output.aux` auxiliary file, and if one uses `bibtex` on it, he will get an `output.bbl` file. On the other hand, it makes it impossible to provide a pre-generated bibliography as a `.bbl` file and be sure that it will be usable whatever the `-jobname`.

This may seem much of a corner case, but it is not because the prominent online tex editor, *overleaf*, sets `-jobname=output` for all its processing. Hence a document `test.tex`, together with a pre-generated `text.bbl` ends up compiling fine *locally* but not working when uploaded to *overleaf*.

The problem is with the definition of the `\bibliography` command in the latex format which unconditionally loads `\jobname.bbl`.  This is inconsistent with the documentation that says

    \bibliography{hfile1,file2, . . . ,fileni} : specifies the bibdata files. Writes a
    \bibdata entry on the .aux file and tries to read in mainfile.bbl.
 
I think that the `\bibliography` command should be modified to get the filename to load from some `\bblfile` macro, presetting it to `\jobname.bbl` so the user can change it if needed. Even better, `latex` should probably preset the `\bblfile` macro to `<mainfile>.bbl` if this file exists or to `\jobname.bbl` if `<mainfile>.bbl` does not exist.  Furthermore, the documentation should be updated accordingly.

**Comment by callegar** [comment]
Indeed... I got mislead by scanning too fast the documentation of the `currfile` package that provides a mainfile related macro. But that works only using the `-recorder` option with latex, actually augmenting the engine features, as you say!

Still, I tend to think that the "extra level of indirection", that is the `\bblfile` macro could be useful, particularly when working with *overleaf*. In fact, explicitly uploading files (which would easily allow renaming the `.bbl` file into whatever needed) is becoming less and less frequent these days, with git or cloud based workflows becoming the norm. In this latter case, setting `\bblfile` is much easier than having two different branches or synchronized workspaces, one for local use and one for pulling/pushing to *overleaf* (either by direct pull/push to overleaf or by pull/push to a github known overleaf or by synchronization with overleaf through dropbox).

With reference to your final considerations about all the other auxiliary files, I really tend to think that `.bbl` is a bit special (and certainly different from `.aux`) for its needing a *database* (e.g. of `.bib` files) that in some cases can be really large and unsuitable for sharing (unless one *extracts* the needed entries on a document by document basis). Furthermore, there is not a single tool by which the `.bbl` can be generated and online tex editors typically provide no way to say which one to use (with some of them that might not even be supported by cloud based tex solutions, such as pybtex).

Your final link about the redefinition of `\jobname` is really interesting. I did not know it could make sense to redefine it from within a document. Still I do not know how reliable it can be to do it with *overleaf* where the site maintainers have made clear that the site logic is built around the assumption that `\jobname` is set to `output`. Incidentally, it is really unclear to me why they cannot rely on the basename of the `.tex` file indicated by the user as the one to run latex on.
